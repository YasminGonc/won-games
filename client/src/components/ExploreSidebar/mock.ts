import { ExploreSideBarProps } from '.'

export const exploreSideBarMock: Pick<ExploreSideBarProps, 'items'> = {
  items: [
    {
      title: 'Price',
      type: 'checkbox',
      fields: [
        'Under $50',
        'Under $100',
        'Under $150',
        'Under $200',
        'Free',
        'Discouted'
      ]
    },
    {
      title: 'Sort by',
      name: 'sort_by',
      type: 'radio',
      fields: ['High to low', 'Low to high']
    },
    {
      title: 'System',
      type: 'checkbox',
      fields: ['Windows', 'Linux', 'MAC']
    },
    {
      title: 'Genre',
      type: 'checkbox',
      fields: ['Action', 'Adventure', 'Simulation']
    }
  ]
}
