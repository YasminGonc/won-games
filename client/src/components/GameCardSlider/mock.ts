import { GameCardProps } from '../GameCard'

const GameCardSliderMock: GameCardProps[] = [
  {
    title: 'Population Zero 1',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: '$ 215,00',
    promotionPrice: '$ 100,00',
    favorite: false,
    hasRibbon: true,
    ribbon: '20% OFF',
    ribbonSize: 'small',
    ribbonColor: 'primary'
  },
  {
    title: 'Population Zero 2',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x141',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 3',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x142',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 4',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x143',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 5',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x144',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 6',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x145',
    price: '$ 215,00'
  }
]

export default GameCardSliderMock
