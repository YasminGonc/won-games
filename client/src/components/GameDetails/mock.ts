import { GameDetailsProps } from '.'

export const GameDetailsMock: GameDetailsProps = {
  developer: 'Gearbox Software',
  publisher: '2K',
  platforms: ['linux', 'windowns', 'mac'],
  realeaseDate: '2020-11-21T23:00:00',
  rating: 'BR0',
  genres: ['Role-playing', 'Narrative']
}
