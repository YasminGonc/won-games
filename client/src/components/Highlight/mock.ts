import { HighlightProps } from '.'

const HighlighMock: HighlightProps = {
  title: 'Read Dead is back',
  subtitle: "Come see John's new adventures",
  backgroundImage: '/img/red-dead-img.png',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
  alignment: 'right'
}

export default HighlighMock
