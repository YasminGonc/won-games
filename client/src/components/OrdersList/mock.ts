import { OrdersListProps } from '.'

export const ordersListMock: OrdersListProps = {
  items: [
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x70',
      title: 'Red Dead Redemption 2',
      price: '$ 215,00',
      downloadLink: 'https://wongames.com/game/download/1',
      paymentInfo: {
        flag: 'mastercard',
        img: '/img/mastercard.png',
        number: '**** **** **** 4326',
        purchaseDate: 'Purchase made on 07/20/2022 at 20:23'
      }
    },
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x71',
      title: 'Red Dead Redemption 2',
      price: '$ 215,00',
      downloadLink: 'https://wongames.com/game/download/2',
      paymentInfo: {
        flag: 'visa',
        img: '/img/visa.png',
        number: '**** **** **** 5423',
        purchaseDate: 'Purchase made on 07/20/2022 at 20:00'
      }
    }
  ]
}
