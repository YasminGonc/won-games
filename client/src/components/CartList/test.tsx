import { renderWithTheme } from '../../utils/tests/helpers'
import { screen } from '@testing-library/react'

import { CartList } from '.'
import { cartListMock } from './mock'

describe('<Footer />', () => {
  it('should render the cart list', () => {
    const { container } = renderWithTheme(<CartList {...cartListMock} />)

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText(cartListMock.total)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
