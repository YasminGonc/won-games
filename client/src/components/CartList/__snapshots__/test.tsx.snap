// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the cart list 1`] = `
.c6 {
  background-color: #EAEAEA;
  color: #030517;
  font-weight: 600;
  padding: 1.6rem 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  color: #F231A5;
}

.c0 {
  background-color: #FAFAFA;
  border-bottom: 0.1rem solid #EAEAEA;
  padding: 2.4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c2 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<div
  class=""
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Red Dead Redemption 2"
          src="https://source.unsplash.com/user/willianjusten/151x70"
        />
      </div>
      <div
        class="c3"
      >
        <h3
          class="c4"
        >
          Red Dead Redemption 2
        </h3>
        <p
          class="c5"
        >
          $ 215,00
        </p>
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Borderlands 3"
          src="https://source.unsplash.com/user/willianjusten/151x71"
        />
      </div>
      <div
        class="c3"
      >
        <h3
          class="c4"
        >
          Borderlands 3
        </h3>
        <p
          class="c5"
        >
          $ 215,00
        </p>
      </div>
    </div>
  </div>
  <div
    class="c6"
  >
    <span>
      Total:
    </span>
    <span
      class="c7"
    >
      $ 430,00
    </span>
  </div>
</div>
`;
